### PR TITLE
Add clarification regarding default values

### LINKED ISSUE #28200: Are there cases where removing a parameter default value would be acceptable?
RE:
> ❌ DISALLOWED: Changing the default value of a property, field, or parameter

Consider the following scenario:
```
public void Foo(int a = 1, int b = 2) { ... }
```

Let's say we want to add a new overload of `Foo` that also accepts `int c`. The following change should (in my understanding) be both binary-compatible and source-compatible:

```
public void Foo(int a, int b) => Foo(a, b, 3);
public void Foo(int a = 1, int b = 2, int c = 3) { ... }
```

(1) Would this be allowed in Microsoft's .NET libraries/runtime?
(2) If it is true that this change does not break either binary or source compatibility, could we add a note about that (I'm happy to file a PR) to this rule for the benefit of those who reference this document to understand breaking changes for .NET generally?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c4b482b0-6dd1-42de-5389-5b0b47823402
* Version Independent ID: 79a4adc9-b2c9-68b7-669e-2bd8d2d44faf
* Content: [Types of breaking changes - .NET](https://docs.microsoft.com/en-us/dotnet/core/compatibility/)
* Content Source: [docs/core/compatibility/index.md](https://github.com/dotnet/docs/blob/main/docs/core/compatibility/index.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @gewarren
* Microsoft Alias: **gewarren**

### PR DESCRIPTION
**Pull request by madelson**
Clarify the compatibility implications of changing or removing parameter default values. 

Fix #28200

## Summary

Describe your changes here.

Fixes #Issue_Number (if available)

### CODE REVIEW

File: docs/core/compatibility/index.md [hunk 1]
```diff
@@ -220,6 +220,10 @@ Changes in this category modify the public surface area of a type. Most of the c
 
 - ❌ **DISALLOWED: Changing the default value of a property, field, or parameter**
 
+  Note that changing or removing a *parameter* default value is not a binary break. Removing a parameter default value is a source break and changing a parameter default value could result in a behavioral break after recompilation.
```
Comment (gewarren) [inline docs/core/compatibility/index.md:223]:
```suggestion
  Changing or removing a *parameter* default value is not a binary break. Removing a parameter default value is a source break, and changing a parameter default value could result in a behavioral break after recompilation.
```

File: docs/core/compatibility/index.md [hunk 2]
```diff
@@ -220,6 +220,10 @@ Changes in this category modify the public surface area of a type. Most of the c
...
+
+  For this reason, removing parameter default values is considered acceptible from a compatibility perspective in the specific case of "moving" those default values to a new method overload in order to eliminate ambiguity. For example, if we have a method `Foo(int a = 1)` and we introduce an overload of `Foo` with 2 optional parameters `a` and `b`, we can preserve compatibility my moving `a`'s default value to the new overload leaving us with `Foo(int a)` and `Foo(int a = 1, int b = 2)`. This will allow `Foo()` to compile.
```
Comment (gewarren) [inline docs/core/compatibility/index.md:225]:
```suggestion
  For this reason, removing parameter default values is acceptable in the specific case of "moving" those default values to a new method overload to eliminate ambiguity. For example, consider an existing method `MyMethod(int a = 1)`. If you introduce an overload of `MyMethod` with two optional parameters `a` and `b`, you can preserve compatibility by moving the default value of `a` to the new overload. Now the two overloads are `MyMethod(int a)` and `MyMethod(int a = 1, int b = 2)`. This pattern allows `MyMethod()` to compile.
```

Comment (gewarren) [reply docs/core/compatibility/index.md:225]: We're discouraged from using "Foo" as a name.

**Review by bartonjs — APPROVED** [review]
With @gewarren 's suggestions, LGTM

**Review by gewarren — APPROVED** [review]

**Pull request merged**